test/functional/lua/ui_spec.lua

```diff
@@ -130,4 +130,44 @@ describe('vim.ui', function()
     end)
 
   end)
+  describe('confirm', function ()
+    it('can choose item', function ()
+      local result = exec_lua[[
+        local opts = {
+            choices = {"Foo", "Bar"},
+	    default = 1,
+        }
+	local ch = nil
+	local on_confirm = function(idx, choice)
+	  ch = choice
+	end
+
+	vim.ui.confirm("Choice?", opts, on_confirm)
+        vim.wait(100, function() return ch ~= nil end)
+	return ch
+      ]]
+      eq('Foo', result)
+    end)
+
+    it('can choose item on nil opt', function()
+      feed(':lua vim.ui.confirm("Choice?", nil, function(idx) result = idx end)<cr>')
+      eq('', eval('v:errmsg'))
+      feed('<cr>')
+      eq(0, exec_lua('return result'))
+    end)
+
+    it('can choose default with cancelreturn', function()
+      feed(':lua vim.ui.confirm("Choice?", {choices = {"Foo", "Bar"}, default = 2}, function(idx, item) result = {idx, item} end)<cr>')
+      eq('', eval('v:errmsg'))
+      feed('<cr>')
+      eq({2, 'Bar'}, exec_lua('return result'))
+    end)
+
+    it('can return nil when aborted with Ctrl-C', function()
+      feed(':lua vim.ui.confirm("Choice?", {choices = {"Foo", "Bar"}, default = 2}, function(idx, item) result = {idx, item} end)<cr>')
+      feed('<c-c>')
+      eq(true, exec_lua('return (nil == result)'))
```

Comment:
Using `feed` is fine for feeding in `<cr>` and `<c-c>`, but no need to use it for `:lua`. Just use `exec_lua` for that.

Reply:
No, `:feed` must be used here. `exec_lua` will hang.

Reply:
Maybe `async_meths.exec_lua` can work though.

Reply:
The latter does not seem to catch `<c-c>` test